File: E2EAppIOS/Sources/Screens/HomeScreen.swift
import SwiftUI

struct HomeScreen: View {
    let onNavigate: (Screen) -> Void

    var body: some View {
        ScreenColumn {
            TaggedText(tag: Tags.homeMarker, text: "E2E ホーム")
            TaggedButton(tag: Tags.navSelector, label: "セレクタ", fillWidth: true) { onNavigate(.selector) }
            TaggedButton(tag: Tags.navInput, label: "テキスト入力", fillWidth: true) { onNavigate(.input) }
            TaggedButton(tag: Tags.navGesture, label: "ジェスチャ", fillWidth: true) { onNavigate(.gesture) }
            TaggedButton(tag: Tags.navScroll, label: "スクロール", fillWidth: true) { onNavigate(.scroll) }
            TaggedButton(tag: Tags.navAsync, label: "非同期表示", fillWidth: true) { onNavigate(.async) }
            TaggedButton(tag: Tags.navDialog, label: "ダイアログ", fillWidth: true) { onNavigate(.dialog) }
            TaggedButton(tag: Tags.navLifecycle, label: "ライフサイクル", fillWidth: true) { onNavigate(.lifecycle) }
            TaggedButton(tag: Tags.navHeal, label: "自己修復", fillWidth: true) { onNavigate(.heal) }
            TaggedButton(tag: Tags.navDiagnostics, label: "診断", fillWidth: true) { onNavigate(.diagnostics) }
        }
    }
}

struct AboutScreen: View {
    var body: some View {
        ScreenColumn {
            TaggedText(tag: Tags.txtAboutMarker, text: "E2E について")
            TaggedText(tag: Tags.txtAboutApp, text: "app=\(AppInfo.appID)")
            TaggedText(tag: Tags.txtAboutVersion, text: "version=\(AppInfo.version)")
        }
    }
}

File: E2EApp/composeApp/src/commonMain/kotlin/com/ftester/e2e/screens/HomeScreen.kt
package com.ftester.e2e.screens

import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.ftester.e2e.Screen
import com.ftester.e2e.Tags
import com.ftester.e2e.ui.ScreenColumn
import com.ftester.e2e.ui.TaggedButton
import com.ftester.e2e.ui.TaggedText

@Composable
fun HomeScreen(onNavigate: (Screen) -> Unit) {
    ScreenColumn(scrollable = true) {
        TaggedText(Tags.HOME_MARKER, "E2E ホーム")
        TaggedButton(Tags.NAV_SELECTOR, "セレクタ", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.SELECTOR) }
        TaggedButton(Tags.NAV_INPUT, "テキスト入力", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.INPUT) }
        TaggedButton(Tags.NAV_GESTURE, "ジェスチャ", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.GESTURE) }
        TaggedButton(Tags.NAV_SCROLL, "スクロール", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.SCROLL) }
        TaggedButton(Tags.NAV_ASYNC, "非同期表示", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.ASYNC) }
        TaggedButton(Tags.NAV_DIALOG, "ダイアログ", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.DIALOG) }
        TaggedButton(Tags.NAV_LIFECYCLE, "ライフサイクル", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.LIFECYCLE) }
        TaggedButton(Tags.NAV_HEAL, "自己修復", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.HEAL) }
        TaggedButton(Tags.NAV_DIAGNOSTICS, "診断", modifier = Modifier.fillMaxWidth()) { onNavigate(Screen.DIAGNOSTICS) }
    }
}
